### PR TITLE
Moved line break validation to separate file

### PR DESCRIPTION
**Pull request by dhuebner**
Also validate following case:
 ```vb
debug = 1
PRINT "Foo"; if debug then  ; REM error here  
PRINT "top_rs_row=",3 
 ``` 

See https://github.com/BBx-Kitchen/bbj-language-server/issues/149#issuecomment-2501204567

### CODE REVIEW

File: bbj-vscode/src/language/validations/line-break-validation.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    const text = textDocument.getText(textRange);
+    return lineStartRegex.test(text);
+}
+
+function hasLinebreakAfter(node: CstNode, textDocument: TextDocument): boolean {
+    const nodeEnd = node.range.end;
+    const textRange: Range = {
+        start: nodeEnd,
+        end: {
+            line: nodeEnd.line + 1,
```
Comment (Lotes) [inline bbj-vscode/src/language/validations/line-break-validation.ts:286]: Thank you for separating out this logic!!!

File: bbj-vscode/test/validation.test.ts [hunk 2]
```diff
@@ -351,10 +351,7 @@ describe('BBj validation', async () => {
         expectNoIssues(validationResult);
     });
 
-    /**
-     * Waiting for answer in #149
-     */
-    test.skip('IF as last child of a compound statement', async () => {
+    test('IF as last child of a compound statement', async () => {
         const validationResult = await validate(`
         debug = 1
```
Comment (Lotes) [inline bbj-vscode/test/validation.test.ts:356]: But why does it work when I add a line break? The FI or ENDIF is still missing.
```
debug = 1
PRINT "Foo";
if debug then
PRINT "top_rs_row=",3
```

**Review by Lotes — APPROVED** [review]
Thanks, I have a question below. But otherwise LGTM, a good improvement.

File: bbj-vscode/test/validation.test.ts (hunk 2 above)
Comment (dhuebner) [reply bbj-vscode/test/validation.test.ts:356]: Thanks for testing!
This is now a multi line IF, not part of #149.
We need to check if missing FI is a valid case or not and open a new issue.

**Pull request merged**